packages/react-components/src/utils/common.ts

```diff
@@ -68,3 +74,94 @@ export const getIconFromUrl = (url: string): JSX.Element | undefined => {
   });
   return icon?.[1];
 };
+
+export type ResearchOutputState = {
+  title: string;
+  description: string;
+  link?: string;
+  type: string;
+  tags: readonly string[];
+  methods: string[];
+  organisms: string[];
+  environments: string[];
+  subtype?: string;
+  labCatalogNumber?: string;
+  teams: ComponentProps<typeof ResearchOutputContributorsCard>['teams'];
+  labs: ComponentProps<typeof ResearchOutputContributorsCard>['labs'];
+  authors: ComponentProps<typeof ResearchOutputContributorsCard>['authors'];
+  identifierType?: ResearchOutputIdentifierType;
+  identifier?: string;
+};
+
+export const isDirty = (
+  {
+    title,
+    description,
+    link,
+    tags,
+    type,
+    methods,
+    organisms,
+    environments,
+    teams,
+    labs,
+    authors,
+    subtype,
+    labCatalogNumber,
+    identifierType,
+    identifier,
+  }: ResearchOutputState,
+  researchOutputData?: ResearchOutputResponse,
+): boolean => {
+  if (researchOutputData) {
+    return (
+      title !== researchOutputData.title ||
+      description !== researchOutputData.description ||
+      link !== researchOutputData.link ||
+      type !== researchOutputData.type ||
+      !equals(methods, researchOutputData.methods) ||
+      !equals(organisms, researchOutputData.organisms) ||
+      !equals(environments, researchOutputData.environments) ||
+      teams?.length !== researchOutputData?.teams.length ||
```

Comment:
I don't think comparing lengths is good enough. We should use .every and check all the ID's are the same. They could be reordered or something.

Reply:
Sent two ID arrays to my equals function as that also checks the order